app/experimenter/static/rapid/contexts/experiment/actions.ts

```diff
@@ -50,3 +50,15 @@ export const updateExperiment = (name: string, value: string | string[]) => (
     },
   });
 };
+
+export const requestReview = () => async (
+  experimentData: ExperimentData,
+  dispatch: React.Dispatch<ExperimentReducerAction>,
+): Promise<void> => {
+  await fetch(`/api/v3/experiments/${experimentData.slug}/request_review/`, {
+    method: "POST",
+  });
+  if (experimentData.slug) {
```

Comment:
Under what circumstances would an experiment _not_ have a slug?
And if there was no slug, what'd be the side effects of the `fetch()` call above on L58-59 which uses `${experimentData.slug}/request_review/` to construct a beautiful request review URL?

Reply:
Slug is `null` before the experiment is created (ie form is filled out but isn't saved yet), so it's optional on the data type:

https://github.com/mozilla/experimenter/blob/main/app/experimenter/static/rapid/types/experiment.ts#L18

so TS enforces checking for it

Reply:
But if `slug` is `null`, wouldn't that make the `await fetch()` call try and set the request_review URL to `/api/v3/experiments//request_review/` (depending on if `null` gets stringified to an empty string or the literal string "null" -- which now I'll have to try verifying out of pure, childlike curioisity).

Reply:
Oh good point we should wrap the whole thing!

Reply:
Or actually, by the time we land on the detail page we expect there to be a slug, so maybe we should just remove the check altogether and let it error out and report since otherwise it's just a totally broken behaviour ie clicking the button and nothing happenign and no error being reported

Reply:
Plus they wouldn't be able to land on the page without a slug

Reply:
Oh no I'm wrong we do have to check it, okay i'll just wrap the whole thing, TS doesn't typecheck the string format it seems.